pkg/termination/write.go

```diff
@@ -30,12 +30,12 @@ const (
 )
 
 // WriteMessage writes the results to the termination message path.
-func WriteMessage(path string, pro []v1beta1.PipelineResourceResult) error {
+func WriteMessage[T v1beta1.TaskRunResult | v1beta1.PipelineResourceResult](path string, pro []T) error {
```

Comment:
This change doesn't require the changes to git-init; can this PR be focused just on just the changes to pkg/termination or at least wait until https://github.com/tektoncd/pipeline/pull/6310 is merged? It could also tackle `termination.ParseMessage`.

Reply:
Any reason to use *generics* here ? 🤔 Not that I am fundamentally against but…

Reply:
Oh because either TaskRunResult or PipelineResourceResult... 😓

Reply:
> Oh because either TaskRunResult or PipelineResourceResult... 😓

trying to migrate off PipelineResourceResult while keeping it least confusing 😿  but seems it could also be confusing. 
I will add more comments in the PR message/commit

Reply:
actually after trying out the unmarshall, i felt generics don't work well with unmarshalls